Highwall and Canopy Height repo setup

alanalutz opened the pull request:
### Description

<!-- Description of the purpose of this PR and what it does. -->

Set up Highwall_Detection and Canopy_Height directories and added code.
This is my initial setup branch to get everything organized and working; I will make a new branch to work on debugging the highwall scripts.

### Changes

<!-- What specific changes were made -->

Canopy_Height:
- added all finalized scripts to /code/ directory
- updated READme.md to explain the purpose of each script and the order in which they are run, and made a note that this code was run in a VM and not ready to be replicated yet

Highwall_Detection
- added working scripts (still in progress) to /code/ directory, and older test versions to /archive/ directory
- added working input data to /inputs/ directory, and created /outputs/ and /temps/ directories
- updated READme.md to match that of MTM_Annual_Extent
- tested get_skeletons.py and segment.py to make sure they work in poetry environment

Overall MTM repo:
- updated pyproject.toml with packages used in highwall scripts (centerline, networkx, shapely, and numpy)
- added contents of inputs, outputs, and temps folders from Highwall_Detection to .gitignore

### Reviewer Checklist

<!-- What would you like the reviewer to check for and how would you like them to check -->

- [ ] Does the structure of both repos look good?
- [ ] Did I do everything right in pyproject.toml and .gitignore?
- [ ] I'm a little confused about changes to the poetry lock file, should they get tracked?

### Associated tickets or issues

<!-- Does this address any specific JIRA tickets or GitHub Issues -->
N/A

### Created, updated, or replaced Files and/or tables

<!-- Are there any files or tables that are created or modified with this PR -->

Lots; see above

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative.py:
```diff
@@ -0,0 +1,112 @@
+for county in counties:
+
+        # Get list of tile IDs from county
+        df = pd.read_csv(f'/home/alanal/gcs/lidar_data/tile_IDs/{county}.csv', header=0)
```
We should update this to read either a file stored in the repo, or GCS

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative.py:
```diff
@@ -0,0 +1,112 @@
+        print(list(duplicates))
+
+        # Create county directory and subdirectories for storing files in bucket
+        dir = f'/home/alanal/gcs/lidar_data/{state}/'
```
This should be restructured to reference a file path in the directory or gcs

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative_repro.py:
```diff
@@ -0,0 +1,148 @@
+        print(list(duplicates))
+
+        # Create county directory and subdirectories for storing files
+        dir = f'/home/alanal/gcs/lidar_data/{state}/'
```
same comment about filepaths

cjthomas730 commented on Canopy_Height/code/post_processing/cog_converter.py:
```diff
@@ -0,0 +1,28 @@
+
+# Paths to input and output files
+input_tif = '/home/alanal/gcs/lidar_data/final_mosaics/complete_dtm_gap_filled_3857.tif'
+output_tif = '/home/alanal/gcs/lidar_data/final_mosaics/complete_dtm_gap_filled_3857_cog.tif'
```
Ideally the input/output paths should point to locations in either GCS or the repo itself

cjthomas730 commented on Canopy_Height/code/post_processing/hole_patch.py:
```diff
@@ -0,0 +1,204 @@
+
+state = 'wv'
+county = 'putnam'
+tile_IDs = ['17SMC20005100', '17SMC20005250', '17SMC20005400', '17SMC20005850', '17SMC18505850', '17SMC17005850', '17SMC17006000', '17SMC17006150', '17SMC15506150', '17SMC14006150', '17SMC12506150', '17SMC11006150', '17SMC11006450', '17SMC09506450', '17SMC06506600', '17SMC08006600', '17SMC08006750', '17SMC08006900', '17SMC08007200', '17SMC08007350', '17SMC08007500', '17SMC09507500', '17SMC0950735 ...
```
For lists like this, or the lists of counties uses in other scripts, un-hardcoding them can make the code shorter, a bit easier to read, and easier to make changes if needed. I'd recommend adding vars like this to the mtm_utils/variables.py

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative.py:
```diff
@@ -0,0 +1,112 @@
+client = storage.Client(project=project_id)
+
+# Access bucket
+bucket_name = 'mountaintop_mining'
```
similar to un-hardcoding variables, you can import bucket name from mtm_utils/variables.py

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative.py:
```diff
@@ -0,0 +1,112 @@
+# Initialize Google Cloud Storage client
+from google.cloud import storage
+project_id = 'skytruth-tech'
```
If you un-hardcode, adding project_id to the variables.py would be helpful

cjthomas730 commented on Canopy_Height/code/processing/decomp_iterative_repro.py:
Same general comments about un-hardcoding

cjthomas730 commented on Canopy_Height/code/processing/mount.py:
same comments about un-hardcoding

cjthomas730 commented on Canopy_Height/code/processing/proc_iterative.py:
```diff
@@ -0,0 +1,226 @@
+    start_time = time.time()
+
+    # Define variables
+    dir = f"/home/alanal/gcs/lidar_data/{state}/{county}/"
```
update path to point to non-local files

cjthomas730 commented on Canopy_Height/code/scraping/scraper.ipynb:
```diff
@@ -0,0 +1,181 @@
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "['2024661NW', '2024669SW', '2024669NW', '2024677SW', '2024677NW', '2024685NW', '2024661NE', '2024669SE', '2024669NE', '2024677SE', '2024677NE', '2024685SE', '2024685NE', '2024693SE', '2024693NE', '2038661NW', '2038669SW', '2038669NW', '2038677SW', '2038677NW', '2038685SW', '2038685NW', '2038693SW', '2038661NE', '2038669SE', '2038669NE', '2038677SE', '2038677NE', '2038685SE', '2038685 ...
```
Another very good place to un-hardcode :)

cjthomas730 commented on Highwall_Detection/archive/christian.py:
Probably worth renaming this script, just so if it's being referenced in the future it's a little clearer what it is, maybe just skelentonize_ct_testing.py

cjthomas730 commented on Highwall_Detection/code/no_spine_extraction.py:
If this code is stil lin-progress, or isn't being used in the overall workflow (since I don't see it in the README.md steps), it might be worth making a testing directory, and keeping the code there

cjthomas730 reviewed (changes requested):
@alanalutz overall everything looks good, but before merging I think it would be worth un-hardcoding some of the variables being used. Happy to chat about this more when you're back!

cjthomas730 commented on poetry.lock:
The packages that are being used for processing (like centerline, networkx, skimage, rasterio, whitebox_workflows, etc. ) in the Highwall Detection and Canopy_Height need to be added to pyproject.toml.

cjthomas730 reviewed (changes requested):
A couple minor changes to be made before merge, but otherwise looks great!

alanalutz commented:
Changes made:
- fixed formatting for the shell sections in the README
- removed compute zone from README
- added section to README detailing the GCS bucket directory structure
- added highwall detection packages to pyproject.toml and tested installation
- added input/output commenting to scripts for clarity

cjthomas730 reviewed (approved):
Everything looks good!
